src/Mod/Sketcher/Gui/DrawSketchHandlerText.h

```diff
@@ -0,0 +1,715 @@
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+
+#ifndef SKETCHERGUI_DrawSketchHandlerText_H
+#define SKETCHERGUI_DrawSketchHandlerText_H
+
+#include <QMap>
+
+#include <Gui/BitmapFactory.h>
+#include <Gui/Notifications.h>
+#include <Gui/Command.h>
+#include <Gui/CommandT.h>
+#include <Gui/InputHint.h>
+
+#include <Mod/Sketcher/App/SketchObject.h>
+
+#include "DrawSketchDefaultWidgetController.h"
+#include "DrawSketchControllableHandler.h"
+
+#include "GeometryCreationMode.h"
+#include "Utils.h"
+#include "CommandConstraints.h"
+
+#include <vector>
+#include <algorithm>
+
+namespace SketcherGui
+{
+
+class DrawSketchHandlerText;
+
+namespace ConstructionMethods
+{
+
+enum class TextConstructionMethod
+{
+    Width,
+    Height,
+    End  // Must be the last one
+};
+
+}
+
+using DSHTextController =
+    DrawSketchDefaultWidgetController<DrawSketchHandlerText,
+                                      /*SelectModeT*/ StateMachines::TwoSeekEnd,
+                                      /*PAutoConstraintSize =*/2,
+                                      /*OnViewParametersT =*/OnViewParameters<4, 4>,  // NOLINT
+                                      /*WidgetParametersT =*/WidgetParameters<0, 0>,  // NOLINT
+                                      /*WidgetCheckboxesT =*/WidgetCheckboxes<0, 0>,  // NOLINT
+                                      /*WidgetComboboxesT =*/WidgetComboboxes<2, 2>,  // NOLINT
+                                      /*WidgetLineEditsT =*/WidgetLineEdits<1, 1>,    // NOLINT
+                                      ConstructionMethods::TextConstructionMethod,
+                                      /*bool PFirstComboboxIsConstructionMethod =*/true>;
+
+using DSHTextControllerBase = DSHTextController::ControllerBase;
+
+using DrawSketchHandlerTextBase = DrawSketchControllableHandler<DSHTextController>;
+
+
+class DrawSketchHandlerText: public DrawSketchHandlerTextBase
+{
+    friend DSHTextController;
+    friend DSHTextControllerBase;
+
+public:
+    explicit DrawSketchHandlerText(ConstructionMethod constrMethod = ConstructionMethod::Width)
+        : DrawSketchHandlerTextBase(constrMethod)
+        , length(0.0)
+        , handleId(0)
+        , text("AstoCAD")

```

Comment:
```suggestion
        , text("Text")

```
can we have some example text which can be translated for different locales?

Reply:
Why not use “Lorem ipsum”? No need to translate and widely known as placeholder